src/components/app/flyouts/settings/platformSettings.js

```diff
@@ -0,0 +1,194 @@
+          {!isValidFile && fileName &&
+            <div className="upload-error-message">
+              <Svg className="upload-error-asterisk" path={svgs.error} alt={t('platformSettings.error')} />
+              {t('platformSettings.uploadError')}
+            </div>
+          }
+          <Section.Content className="platform-section-description show-line-breaks">{t('platformSettings.logoDescription')}</Section.Content>
+        </div>
+        <Section.Content className="name-input-container">
+          <div className="section-subtitle">{t('platformSettings.applicationName')}</div>
+          <FormControl type="text" className="name-input long" classForLabel="description"
+            isEdit={true} placeholder={this.state.currentApplicationName} onChange={this.onNameChange} />
+        </Section.Content>
+      </div>
+    );
+  }
+
+  render() {
+    const { t } = this.props;
+    const { logoIsDefault, validating } = this.state;
+    return (
+      <Section.Container collapsable={true} className="setting-section">
+        <Section.Header>{t('platformSettings.nameAndLogo')}</Section.Header>
+        <Section.Content>{t('platformSettings.nameLogoDescription')}</Section.Content>
+        {this.state.stillInitializing ?
+          <Indicator size='medium' />
+          :
+          <Section.Content>
+            {this.state.edit ?
+              validating ?
+                <div className="upload-logo-name-container">
+                  <Indicator size='small' />
+                </div>
+                :
+                this.renderUploadContainer()
+              :
+              <div>
+                <div className="current-logo-container">
+                  <div className="current-logo-name">
+                    <div className="current-logo">
+                      {logoIsDefault ?
+                        this.renderSvgLogo(this.state.currentLogo)
+                        : <img className="current-logo" src={this.state.currentLogo} alt={t('platformSettings.currentLogo')} />
+                      }
+                    </div>
+                    <div className="name-container">{this.state.currentApplicationName}</div>
+                  </div>
+                  <div className="edit-button-div">
+                    <Btn svg={svgs.edit} onClick={this.enableEdit} className="edit-button">{t('platformSettings.edit')}</Btn>
+                  </div>
+                </div>
+              </div>
+            }
+          </Section.Content>
+        }
+      </Section.Container>
+    );
+  }
+
+  enableEdit = () => {
+    this.setState({
+      edit: true
+    });
+  }
+
+  onNameChange = (e) => {
+    var { value } = e.target;
+    if (value.length === 0) {
+      value = undefined;
+    }
+    this.props.onNameChange(value);
+  };
+
+  onUpload = (e) => {
+    var file = e.target.files[0];
+    this.setState({
+      validating: true,
+      validFile: false
+    });
+    if (file !== undefined && this.isValidLogoFile(file)) {
```

Comment:
what if file is null `if (file && this.isValidLogoFile(file))` would work better. Though in theory isValidLogoFile should check for undefined or null.